Sources/ArcGISToolkit/Components/Offline/OfflineMapAreasView.swift

```diff
@@ -104,6 +116,33 @@ public struct OfflineMapAreasView: View {
         }
         .frame(maxWidth: .infinity)
     }
+    
+    @ViewBuilder private var emptyOfflinePreplannedMapAreasView: some View {
+        VStack(alignment: .center) {
+            Text("No offline map areas")
+                .bold()
+            Text("You don't have any downloaded offline map areas yet.")
+                .font(.subheadline)
+                .foregroundStyle(.secondary)
+        }
+        .frame(maxWidth: .infinity)
+    }
+    
+    @ViewBuilder private func errorView(_ error: Error) -> some View {
+        VStack(alignment: .center) {
+            Image(systemName: "exclamationmark.circle")
+                .imageScale(.large)
+                .foregroundStyle(.red)
+            Text(error.localizedDescription)
+        }
+        .frame(maxWidth: .infinity)
+    }
+    
+    /// Makes the preplanned map models.
+    private func makePreplannedMapModels() async {
+        await mapViewModel.makePreplannedMapModels()
+        await mapViewModel.makeOfflinePreplannedMapModels()
```

Comment:
This isn't ideal - each time the `makePreplannedMapModels` method is called, both the method to load areas from the server, and the method to load mmpks from disk are called, but only 1 group is used. I think the problem stems in that we used the `mapViewModel.preplannedMapModels` to determine what UI is shown, which favors remote areas over local ones.

A thought that we may need some discussion:

- Instead of using `mapViewModel.preplannedMapModels`, split the `preplannedMapAreaViews` into 2 views: `onlinePreplannedMapAreaViews` and `offlinePreplannedMapAreaViews`. Then, call `try await offlineMapTask.preplannedMapAreas` or use `NWPath` to determine which view to show

Reply:
Great suggestion. There are now two views for when the device is online or offline determined by `NWPath`, and only the respective models are created instead of both sets. I verified that the correct views are shown when the internet connection is toggled on and off on a device.